src/Revenue/Repositories/Revenue.php

```diff
@@ -0,0 +1,90 @@
+<?php
+namespace Give\Revenue\Repositories;
+
+use InvalidArgumentException;
+
+/**
+ * Class Revenue
+ * @package Give\Revenue\Repositories
+ *
+ * Use this class to get data from "give_revenue" table.
+ *
+ * @since 2.9.0
+ */
+class Revenue {
+	/**
+	 * Revenue constructor
+	 */
+	public function constructor() {
+	}
+
+	/**
+	 * Insert revenue.
+	 *
+	 * @since 2.9.0
+	 *
+	 * @param  array  $revenueData
+	 *
+	 * @return bool|int
+	 */
+	public function insert( $revenueData ) {
+		global $wpdb;
+
+		$this->validateDataFormInsertion( $revenueData );
+
+		/**
+		 * Filter data name for revenue table.
+		 *
+		 * @since 2.9.0
+		 */
+		$revenueData = apply_filters(
+			'give_revenue_data_for_insertion',
+			$revenueData
+		);
+
+		return $wpdb->insert(
+			$wpdb->give_revenue,
+			$revenueData,
+			$this->getPlaceholderForPrepareQuery( $revenueData )
+		);
+	}
+
+	/**
+	 * Validate revenue data for insertion.
+	 *
+	 * @since 2.9.0
+	 *
+	 * @param array $array
+	 */
+	private function validateDataFormInsertion( $array ) {
```

Comment:
Typo in the method name?

Reply:
fixed: https://github.com/impress-org/givewp/compare/e05f93d70f3c...b140fdebe02d